samples/highcharts/yaxis/type-log-negative/demo.js

```diff
@@ -37,6 +41,90 @@
             };
         }
     });
+
+    // Add support for negative axis values to the tick positioning function.
+    H.wrap(logAdditions, 'getLogTickPositions', function (
+        proceed,
+        interval,
+        min,
+        max,
+        minor
+    ) {
+        if (
+            !this.axis.options.custom.allowNegativeLog ||
+            interval >= 0.5 ||
+            interval < 0.08
+        ) {
+            return proceed.call(this, interval, min, max, minor);
+        }
```

Comment:
Anyway, I've slightly improved this solution using a wrapper, removing as much core code repetition as possible and reducing the error-proneness associated with future Highcharts updates.